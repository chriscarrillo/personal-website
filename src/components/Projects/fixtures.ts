import klawWeb from 'assets/images/klaw-web.png'
import napkin from 'assets/images/napkin.png'
import portfolio from 'assets/images/portfolio.png'
import reciplease from 'assets/images/reciplease.png'
import {Project} from 'models'

/**
 * Projects.
 */
export const projects: Project[] = [
  {
    description: `
      An e-commerce, business website for a life and ennegram coach in which customers can buy
      typing sessions and learn more about the enneagram.`,
    image: '',
    link: '',
    name: 'Kenna Ledbetter Coaching',
    repositoryLink: '',
    technologies: ['React', 'TypeScript', 'Commerce.js', 'Stripe'],
  },
  {
    description: `
      A web application to create meal plans and add meals for those corresponding plans.`,
    image: '',
    link: '',
    name: 'Swole Your Rolls',
    repositoryLink: 'https://github.com/chriscarrillo/swole-your-rolls',
    technologies: ['React', 'TypeScript', 'Firebase'],
  },
  {
    description: 'A personal website that showcases my projects and shares more info about me.',
    image: portfolio,
    link: '',
    name: 'Portfolio',
    repositoryLink: 'https://github.com/chriscarrillo/personal-website',
    technologies: ['React', 'TypeScript', 'Bootstrap'],
  },
  {
    description: `
      An educational website designed to help STEM students and hobbyists learn about
      robotics and programming.`,
    image: klawWeb,
    link: 'https://klaw-web.herokuapp.com/',
    name: 'KLAW-Web',
    repositoryLink: 'https://github.com/chriscarrillo/KLAW-Web',
    technologies: ['Angular', 'JavaScript', 'Blockly', 'Flask', 'Python', 'WebGL'],
  },
  {
    description: `
      Android GUI Calculator for frequently used calculations. This includes lifestyle,
      finance, school, and conversion calculators.`,
    image: napkin,
    link: '',
    name: 'Napkin',
    repositoryLink: 'https://github.com/chriscarrillo/Napkin',
    technologies: ['Java'],
  },
  {
    description: `
      A recipe finder website that allows users to quickly and easily find recipes that fit into
      their dietary restrictions and/or what they have available to them.`,
    image: reciplease,
    link: '',
    name: 'Reciplease',
    repositoryLink: 'https://github.com/chriscarrillo/Reciplease',
    technologies: ['PHP', 'JavaScript'],
  },
]
