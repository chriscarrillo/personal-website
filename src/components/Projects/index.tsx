import klawWeb from 'assets/images/klaw-web.png'
import {Project} from 'models'
import React from 'react'
import {Card, CardDeck, Col, Container, Row} from 'react-bootstrap'
import {FaExternalLinkAlt, FaGithub} from 'react-icons/fa'
import {MdDeveloperMode} from 'react-icons/md'

const projects: Project[] = [
  {
    description: 'A personal website that showcases my projects and shares more info about me.',
    image: '',
    link: '',
    name: 'Portfolio',
    repositoryLink: 'https://github.com/chriscarrillo/personal-website',
    technologies: ['React', 'TypeScript'],
  },
  {
    description: `
      A web application to create meal plans and add meals for those corresponding plans.`,
    image: '',
    link: '',
    name: 'Swole Your Rolls',
    repositoryLink: 'https://github.com/chriscarrillo/swole-your-rolls',
    technologies: ['React', 'TypeScript', 'Firebase'],
  },
  {
    description: `
      An e-commerce, business website for a life and ennegram coach in which customers can buy
      typing sessions and learn more about the enneagram.`,
    image: '',
    link: '',
    name: 'Kenna Ledbetter Coaching',
    repositoryLink: '',
    technologies: ['React', 'TypeScript', 'Commerce.js'],
  },
  {
    description: `
      An educational website designed to help STEM students and hobbyists learn about
      robotics and programming.`,
    image: klawWeb,
    link: 'https://klaw-web.herokuapp.com/',
    name: 'KLAW-Web',
    repositoryLink: 'https://github.com/chriscarrillo/KLAW-Web',
    technologies: ['Angular', 'JavaScript', 'Blockly', 'Flask', 'Python', 'WebGL'],
  },
  {
    description: 'Android GUI Calculator for frequently used calculations.',
    image: '',
    link: '',
    name: 'Napkin',
    repositoryLink: 'https://github.com/chriscarrillo/Napkin',
    technologies: ['Java'],
  },
  {
    description: `
      A recipe finder website that allows users to quickly and easily find recipes that fit into
      their dietary restrictions and/or what they have available to them.`,
    image: '',
    link: '',
    name: 'Reciplease',
    repositoryLink: 'https://github.com/chriscarrillo/Reciplease',
    technologies: ['PHP', 'Python'],
  },
]

/**
 * Projects section.
 * @return React components
 */
export const Projects: React.FC = () => {
  const projectsRow = projects.map(project => {
    const projectImage =
      project.image.length === 0 ? (
        <div className="p-2 text-center text-light bg-dark">
          <Card.Title>
            <MdDeveloperMode size={40} />
          </Card.Title>
          <Card.Text>This project is still in development.</Card.Text>
        </div>
      ) : (
        <Card.Img alt={project.name} src={project.image} />
      )
    const technologies = project.technologies.map((technology, index) => (
      <span key={technology}>
        {technology}
        {index < project.technologies.length - 1 ? ' â€” ' : ''}
      </span>
    ))
    const repoLink = project.repositoryLink !== undefined && project.repositoryLink.length > 0 && (
      <a className="text-dark pr-2" href={project.repositoryLink} rel="noreferrer" target="_blank">
        <FaGithub className="social-link" size={26}>
          GitHub
        </FaGithub>
      </a>
    )
    const projectLink = project.link !== undefined && project.link.length > 0 && (
      <a
        className="text-dark"
        href={project.link}
        rel="noreferrer"
        target="_blank"
        title="See it live!"
      >
        <FaExternalLinkAlt className="social-link" size={26}>
          {project.name}
        </FaExternalLinkAlt>
      </a>
    )

    return (
      <Col key={project.name} as={CardDeck} className="py-2" md={4} sm={12}>
        <Card>
          <Card.Header className="d-flex justify-content-between">
            <span className="h4 family-semi">{project.name}</span>
            {projectLink}
          </Card.Header>
          {projectImage}
          <Card.Body>
            <Card.Text>{project.description}</Card.Text>
          </Card.Body>
          <Card.Footer>
            <span className="d-block family-semi">Technologies:</span>
            {technologies}
            <div className="pt-1">
              {repoLink}
              {projectLink}
            </div>
          </Card.Footer>
        </Card>
      </Col>
    )
  })

  return (
    <section className="bg-light py-4">
      <Container fluid className="d-flex flex-column align-items-center justify-content-center">
        <h1 className="text-dark text-uppercase family-semi pb-2 mb-0">Projects</h1>
        <Row className="justify-content-center">{projectsRow}</Row>
      </Container>
    </section>
  )
}
