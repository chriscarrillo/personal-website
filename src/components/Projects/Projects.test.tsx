import {describe, expect, it} from '@jest/globals'
import {render, screen} from '@testing-library/react'
import {Projects} from 'components/Projects'
import React from 'react'
import renderer from 'react-test-renderer'
import {projects} from './fixtures'

describe('<Projects />', () => {
  it('should render as expected', () => {
    const tree = renderer.create(<Projects />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render heading', async () => {
    const {findByText} = render(<Projects />)
    await expect(findByText('Projects')).toBeTruthy()
  })

  it('should render all project cards', async () => {
    render(<Projects />)
    const cards = await screen.findAllByTestId('projectCard')
    expect(cards).toBeTruthy()
    expect(cards).toHaveLength(projects.length)
  })
})
