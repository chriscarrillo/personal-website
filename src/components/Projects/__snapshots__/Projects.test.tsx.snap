// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Projects /> should render as expected 1`] = `
<section
  className="bg-light py-4"
>
  <div
    className="d-flex flex-column align-items-center justify-content-center container"
  >
    <h1
      className="text-dark text-uppercase family-semi pb-2 mb-0"
      data-aos="flip-up"
    >
      Projects
    </h1>
    <div
      className="justify-content-center row"
    >
      <div
        className="py-2 col-md-4 col-sm-12 card-deck"
      >
        <div
          className="card"
          data-aos="flip-left"
          data-testid="projectCard"
        >
          <div
            className="d-flex justify-content-between card-header"
          >
            <span
              className="h5 family-semi"
            >
              Kenna Ledbetter Coaching
            </span>
          </div>
          <div
            className="p-2 text-center text-light bg-dark"
          >
            <div
              className="card-title h5"
            >
              <svg
                fill="currentColor"
                height={40}
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width={40}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7 5h10v2h2V3c0-1.1-.9-1.99-2-1.99L7 1c-1.1 0-2 .9-2 2v4h2V5zm8.41 11.59L20 12l-4.59-4.59L14 8.83 17.17 12 14 15.17l1.41 1.42zM10 15.17L6.83 12 10 8.83 8.59 7.41 4 12l4.59 4.59L10 15.17zM17 19H7v-2H5v4c0 1.1.9 2 2 2h10c1.1 0 2-.9 2-2v-4h-2v2z"
                />
              </svg>
            </div>
            <p
              className="card-text"
            >
              This project is still in development.
            </p>
          </div>
          <div
            className="card-body"
          >
            <p
              className="card-text"
            >
              
      An e-commerce, business website for a life and ennegram coach in which customers can buy
      typing sessions and learn more about the enneagram.
            </p>
          </div>
          <div
            className="card-footer"
          >
            <span
              className="d-block family-semi"
            >
              Technologies:
            </span>
            <span>
              React
              , 
            </span>
            <span>
              TypeScript
              , 
            </span>
            <span>
              Commerce.js
              , 
            </span>
            <span>
              Stripe
              
            </span>
            <div
              className="pt-1"
            />
          </div>
        </div>
      </div>
      <div
        className="py-2 col-md-4 col-sm-12 card-deck"
      >
        <div
          className="card"
          data-aos="flip-left"
          data-testid="projectCard"
        >
          <div
            className="d-flex justify-content-between card-header"
          >
            <span
              className="h5 family-semi"
            >
              Swole Your Rolls
            </span>
          </div>
          <div
            className="p-2 text-center text-light bg-dark"
          >
            <div
              className="card-title h5"
            >
              <svg
                fill="currentColor"
                height={40}
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width={40}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7 5h10v2h2V3c0-1.1-.9-1.99-2-1.99L7 1c-1.1 0-2 .9-2 2v4h2V5zm8.41 11.59L20 12l-4.59-4.59L14 8.83 17.17 12 14 15.17l1.41 1.42zM10 15.17L6.83 12 10 8.83 8.59 7.41 4 12l4.59 4.59L10 15.17zM17 19H7v-2H5v4c0 1.1.9 2 2 2h10c1.1 0 2-.9 2-2v-4h-2v2z"
                />
              </svg>
            </div>
            <p
              className="card-text"
            >
              This project is still in development.
            </p>
          </div>
          <div
            className="card-body"
          >
            <p
              className="card-text"
            >
              
      A web application to create meal plans and add meals for those corresponding plans.
            </p>
          </div>
          <div
            className="card-footer"
          >
            <span
              className="d-block family-semi"
            >
              Technologies:
            </span>
            <span>
              React
              , 
            </span>
            <span>
              TypeScript
              , 
            </span>
            <span>
              Firebase
              
            </span>
            <div
              className="pt-1"
            >
              <a
                className="text-dark pr-2"
                href="https://github.com/chriscarrillo/swole-your-rolls"
                rel="noreferrer"
                target="_blank"
              >
                <svg
                  className="social-link"
                  fill="currentColor"
                  height={26}
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 496 512"
                  width={26}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
                  />
                </svg>
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="py-2 col-md-4 col-sm-12 card-deck"
      >
        <div
          className="card"
          data-aos="flip-left"
          data-testid="projectCard"
        >
          <div
            className="d-flex justify-content-between card-header"
          >
            <span
              className="h5 family-semi"
            >
              Portfolio
            </span>
          </div>
          <img
            alt="Portfolio"
            className="card-img"
            src="portfolio.png"
          />
          <div
            className="card-body"
          >
            <p
              className="card-text"
            >
              A personal website that showcases my projects and shares more info about me.
            </p>
          </div>
          <div
            className="card-footer"
          >
            <span
              className="d-block family-semi"
            >
              Technologies:
            </span>
            <span>
              React
              , 
            </span>
            <span>
              TypeScript
              , 
            </span>
            <span>
              Bootstrap
              
            </span>
            <div
              className="pt-1"
            >
              <a
                className="text-dark pr-2"
                href="https://github.com/chriscarrillo/personal-website"
                rel="noreferrer"
                target="_blank"
              >
                <svg
                  className="social-link"
                  fill="currentColor"
                  height={26}
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 496 512"
                  width={26}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
                  />
                </svg>
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="py-2 col-md-4 col-sm-12 card-deck"
      >
        <div
          className="card"
          data-aos="flip-left"
          data-testid="projectCard"
        >
          <div
            className="d-flex justify-content-between card-header"
          >
            <span
              className="h5 family-semi"
            >
              KLAW-Web
            </span>
            <a
              className="text-dark"
              href="https://klaw-web.herokuapp.com/"
              rel="noreferrer"
              target="_blank"
              title="See it live!"
            >
              <svg
                className="social-link"
                fill="currentColor"
                height={26}
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 512 512"
                width={26}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                />
              </svg>
            </a>
          </div>
          <img
            alt="KLAW-Web"
            className="card-img"
            src="klaw-web.png"
          />
          <div
            className="card-body"
          >
            <p
              className="card-text"
            >
              
      An educational website designed to help STEM students and hobbyists learn about
      robotics and programming.
            </p>
          </div>
          <div
            className="card-footer"
          >
            <span
              className="d-block family-semi"
            >
              Technologies:
            </span>
            <span>
              Angular
              , 
            </span>
            <span>
              JavaScript
              , 
            </span>
            <span>
              Blockly
              , 
            </span>
            <span>
              Flask
              , 
            </span>
            <span>
              Python
              , 
            </span>
            <span>
              WebGL
              
            </span>
            <div
              className="pt-1"
            >
              <a
                className="text-dark pr-2"
                href="https://github.com/chriscarrillo/KLAW-Web"
                rel="noreferrer"
                target="_blank"
              >
                <svg
                  className="social-link"
                  fill="currentColor"
                  height={26}
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 496 512"
                  width={26}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
                  />
                </svg>
              </a>
              <a
                className="text-dark"
                href="https://klaw-web.herokuapp.com/"
                rel="noreferrer"
                target="_blank"
                title="See it live!"
              >
                <svg
                  className="social-link"
                  fill="currentColor"
                  height={26}
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 512 512"
                  width={26}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                  />
                </svg>
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="py-2 col-md-4 col-sm-12 card-deck"
      >
        <div
          className="card"
          data-aos="flip-left"
          data-testid="projectCard"
        >
          <div
            className="d-flex justify-content-between card-header"
          >
            <span
              className="h5 family-semi"
            >
              Napkin
            </span>
          </div>
          <img
            alt="Napkin"
            className="card-img"
            src="napkin.png"
          />
          <div
            className="card-body"
          >
            <p
              className="card-text"
            >
              
      Android GUI Calculator for frequently used calculations. This includes lifestyle,
      finance, school, and conversion calculators.
            </p>
          </div>
          <div
            className="card-footer"
          >
            <span
              className="d-block family-semi"
            >
              Technologies:
            </span>
            <span>
              Java
              
            </span>
            <div
              className="pt-1"
            >
              <a
                className="text-dark pr-2"
                href="https://github.com/chriscarrillo/Napkin"
                rel="noreferrer"
                target="_blank"
              >
                <svg
                  className="social-link"
                  fill="currentColor"
                  height={26}
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 496 512"
                  width={26}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
                  />
                </svg>
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="py-2 col-md-4 col-sm-12 card-deck"
      >
        <div
          className="card"
          data-aos="flip-left"
          data-testid="projectCard"
        >
          <div
            className="d-flex justify-content-between card-header"
          >
            <span
              className="h5 family-semi"
            >
              Reciplease
            </span>
          </div>
          <img
            alt="Reciplease"
            className="card-img"
            src="reciplease.png"
          />
          <div
            className="card-body"
          >
            <p
              className="card-text"
            >
              
      A recipe finder website that allows users to quickly and easily find recipes that fit into
      their dietary restrictions and/or what they have available to them.
            </p>
          </div>
          <div
            className="card-footer"
          >
            <span
              className="d-block family-semi"
            >
              Technologies:
            </span>
            <span>
              PHP
              , 
            </span>
            <span>
              JavaScript
              
            </span>
            <div
              className="pt-1"
            >
              <a
                className="text-dark pr-2"
                href="https://github.com/chriscarrillo/Reciplease"
                rel="noreferrer"
                target="_blank"
              >
                <svg
                  className="social-link"
                  fill="currentColor"
                  height={26}
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 496 512"
                  width={26}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
                  />
                </svg>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
